src/main/java/flaxbeard/immersivepetroleum/common/blocks/metal/TileEntityPumpjack.java

```diff
@@ -172,7 +198,7 @@ public void update(boolean consumePower)
 		int consumed = IPConfig.Extraction.pumpjack_consumption;
 		int extracted = consumePower ? energyStorage.extractEnergy(consumed, true) : consumed;
 				
-		if (extracted >= consumed && canExtract() && !this.isRSDisabled())
+		if (extracted >= consumed && canExtract() && !this.isRSDisabled() && hasPipe())
```

Comment:
I was thinking the same thing about the performance. What if I made it another config option? Five seconds sounds like a good default, and then if anyone needs to raise it because they have a ridiculous number of jacks or something they still could. Good idea about the canExtract(), having a pipe is really just an optional check for that same behavior.